PersonDataRequest/PersonDataRequest/Controller/DataViewController.swift

```diff
@@ -0,0 +1,67 @@
+//
+//  DataViewController.swift
+//  PersonDataRequest
+//
+//  Created by Ivan on 25.01.2024.
+//
+
+import UIKit
+
+class DataViewController: UIViewController {
+    
+    var selectedName: String?
+    
+    @IBOutlet weak var nameLabel: UILabel!
+    @IBOutlet weak var genderLabel: UILabel!
+    @IBOutlet weak var probabilityLabel: UILabel!
+    @IBOutlet weak var ageLabel: UILabel!
+    
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        nameLabel.text = selectedName
+        
+        netRequest()
+    }
+    
+    func netRequest() {
+        guard let selectedName = selectedName else { return }
+        let genderUrl = "https://api.genderize.io/?name=\(selectedName)"
+        let ageUrl = "https://api.agify.io/?name=\(selectedName)"
+        
+        let apiHelper = ApiHelper()
+        
+        apiHelper.makeRequest(urlString: genderUrl, t: GenderData.self) { genderData in
```

Comment:
You need to add `[weak self]` before `genderData`, this is a rule for closures where `self` is used

Reply:
I added [weak self] before genderData and made self? in the next row